src/objects.h

```diff
@@ -1246,7 +1247,7 @@ class PromotedTrack : public HeapObject {
   // We only want to traverse the newly-promoted objects explicitly.
   uword size() {
     ASSERT(class_tag() == PROMOTED_TRACK_TAG);
-    return HEADER_SIZE;
+    return end() - _raw();
```

Comment:
When traversing the heap we want to skip the newly promoted objects that are inside the PromotionTrack object.